### PR TITLE
Add docs to string methods in `SymbolLiteral` and `MacroId`

### PR DESCRIPTION
**Pull request by MakeNowJust**
Because their methods delegates to `StringLiteral`. In fact, `SymbolLiteral#underscore` and `MacroId#underscore` are missing in document. It should be shared.

In details, it uses a file `private` macro named `def_string_methods`.

Thank you.

### CODE REVIEW

**Comment by MakeNowJust** [comment]
Follow #9269 changes. `#titleize` is added.

**Comment by MakeNowJust** [comment]
ping

**Comment by j8r** [comment]
Any reasons against using a generic module instead of a macro?

**Comment by MakeNowJust** [comment]
Because they don't include any
common module in fact.

**Comment by j8r** [comment]
I don't quite understand, I mean having something like `include StringMethods(MacroId)` instead of def_string_methods MacroId`. Do you say it won't make sense?

**Comment by asterite** [comment]
It won't make sense. Such hierarchy doesn't exist in the compiler. The type doesn't exist in macros.

**Comment by MakeNowJust** [comment]
Repeat from the above comment, current documentation misses `SymbolLiteral#underscore` and `MacroId#underscore`, so this PR fixes this documentation bug. It is indirect, but I think it is more reasonable than adding each methods to their definitions.

**Review by straight-shoota — APPROVED** [review]

File: src/compiler/crystal/macros.cr [hunk 1]
```diff
@@ -1,3 +1,118 @@
...
+  def =~(range : RegexLiteral) : BoolLiteral
+  end
+
+  # Similar to `String#>`
+  def >(other : StringLiteral | MacroId) : BoolLiteral
+  end
+
+  # Similar to `String#<`
+  def <(other : StringLiteral | MacroId) : BoolLiteral
+  end
```
Comment (oprypin) [inline src/compiler/crystal/macros.cr:25]: `SymbolLiteral` doesn't have `<` `>` methods.

The rest seems to be good.

**Pull request merged**